src/net/sf/mpxj/primavera/PrimaveraPMFileReader.java

```diff
@@ -639,42 +639,45 @@ private void updateDates(Task parentTask)
          {
             updateDates(task);
 
-            if (actualStartDate == null || DateHelper.compare(actualStartDate, task.getActualStart()) > 0)
+            // the child tasks can have null dates (e.g. for nested wbs elements with no task children) so we
+            // still must protect against some children having null dates
+
+            if (actualStartDate == null || (task.getActualStart() != null && DateHelper.compare(actualStartDate, task.getActualStart()) > 0))
             {
                actualStartDate = task.getActualStart();
             }
 
-            if (actualFinishDate == null || DateHelper.compare(actualFinishDate, task.getActualFinish()) < 0)
+            if (actualFinishDate == null || (task.getActualFinish() != null && DateHelper.compare(actualFinishDate, task.getActualFinish()) < 0))
```

Comment:
Ok, went ahead and added the null-safe `DateHelper.min/max` which simplifies the date rollup code a lot and makes sure that bug is fixed. However, the three places I found similar date rollup logic were all rolling up slightly different sets of dates so I didn't factor out the `updateDates` functions into a single common implementation.